torchreid/integration/nncf/compression.py

```diff
@@ -0,0 +1,247 @@
+        state_dict = checkpoint['state_dict']
+    else:
+        raise RuntimeError(
+            'No state_dict found in checkpoint file {}'.format(filename))
+    _ = load_state(model, state_dict, strict)
+    return checkpoint
+
+def wrap_nncf_model(model, cfg, datamanager_for_init,
+                    checkpoint_path=None):
+    # Note that we require to import it here to avoid cyclic imports when import get_no_nncf_trace_context_manager
+    # from mobilenetv3
+    from torchreid.data.transforms import build_inference_transform
+    from torchreid.utils import read_json
+
+    if not (datamanager_for_init or checkpoint_path):
+        raise RuntimeError(f'One of datamanager_for_init or checkpoint_path should be set: '
+                           f'datamanager_for_init={datamanager_for_init} checkpoint_path={checkpoint_path}')
+
+    import nncf
+    from nncf import (NNCFConfig, create_compressed_model,
+                      register_default_init_args)
+    from nncf.initialization import InitializingDataLoader
+    from nncf.dynamic_graph.input_wrapping import nncf_model_input
+    from nncf.dynamic_graph.trace_tensor import TracedTensor
+
+    # Taking the part, related to NNCF, from the model config.
+    # Please, note that at the moment the part should be a dict of the following form:
+    # ```
+    # nncf = {
+    #     'nncf_config_path': '...' #this is the path to a json file with NNCF config dict itself
+    #     'coeff_decrease_lr_for_nncf': <float value> # this is a coefficient to decrease the LR for NNCF training
+    # }
+    # ```
+    # -- in case if some section of this NNCF part is absent the default values will be used
+    nncf_external_config = cfg.get('nncf')
+
+    nncf_metainfo = _get_nncf_metainfo_from_checkpoint(checkpoint_path)
+    if nncf_metainfo and nncf_metainfo.get('nncf_compression_enabled'):
+        nncf_config_data = nncf_metainfo['nncf_config']
+        datamanager_for_init = None
+        print(f'Read NNCF metainfo from the checkpoint: nncf_metainfo=\n{pformat(nncf_metainfo)}')
+    else:
+        checkpoint_path = None # it is non-NNCF model
+
+        nncf_config_data = None
+        if nncf_external_config and nncf_external_config.get('nncf_config_path'):
+            nncf_config_path = nncf_external_config.get('nncf_config_path')
+            nncf_config_data = read_json(nncf_config_path)
+            print(f'Read nncf config from the NNCF config file {nncf_config_path}:\n nncf_config=\n{pformat(nncf_config_data)}')
+        if nncf_config_data is None:
+            print('Cannot read nncf_config from config file')
+
+    if datamanager_for_init and checkpoint_path:
+        raise RuntimeError(f'Only ONE of datamanager_for_init or checkpoint_path should be set: '
+                           f'datamanager_for_init={datamanager_for_init} checkpoint_path={checkpoint_path}')
+    if not nncf_config_data:
+        print('Using the default NNCF int8 quantization config')
+        nncf_config_data = {
```

Comment:
Seems like the same structure appears in `add_nncf_metainfo_to_checkpoint.py`. Could we keep it here and import in other cases?

Reply:
Ok, I will move it to a special file.

Reply:
Moved it to a separate function `get_default_nncf_compression_config` in `torchreid/integration/nncf/compression.py`